contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -0,0 +1,138 @@
+use soroban_sdk::{Address, BytesN, Env, Vec};
+use stellar_axelar_std::interfaces::{CustomMigratableInterface, UpgradableClient};
+
+use crate::error::ContractError;
+use crate::{deployer, storage, InterchainTokenService};
+
+pub mod legacy_storage {
+    use soroban_sdk::{contracttype, Address, BytesN, String};
+    use stellar_axelar_std::contractstorage;
+
+    use crate::storage::TokenIdConfigValue;
+
+    #[contractstorage]
+    enum DataKey {
+        #[instance]
+        #[value(Address)]
+        Gateway,
+
+        #[instance]
+        #[value(Address)]
+        GasService,
+
+        #[instance]
+        #[value(String)]
+        ChainName,
+
+        #[instance]
+        #[value(String)]
+        ItsHubAddress,
+
+        #[instance]
+        #[value(Address)]
+        NativeTokenAddress,
+
+        #[instance]
+        #[value(BytesN<32>)]
+        InterchainTokenWasmHash,
+
+        #[instance]
+        #[value(BytesN<32>)]
+        TokenManagerWasmHash,
+
+        #[persistent]
+        #[status]
+        TrustedChain { chain: String },
+
+        #[persistent]
+        #[value(TokenIdConfigValue)]
+        TokenIdConfig { token_id: BytesN<32> },
+
+        #[persistent]
+        #[value(i128)]
+        FlowLimit { token_id: BytesN<32> },
+
+        #[temporary]
+        #[value(i128)]
+        FlowOut { flow_key: FlowKey },
+
+        #[temporary]
+        #[value(i128)]
+        FlowIn { flow_key: FlowKey },
+    }
+
+    #[contracttype]
+    #[derive(Clone, Debug)]
+    pub struct FlowKey {
+        pub token_id: BytesN<32>,
+        pub epoch: u64,
+    }
+}
+
+impl CustomMigratableInterface for InterchainTokenService {
+    type MigrationData = (
+        BytesN<32>, /* new_wasm_hash */
+        Vec<(BytesN<32> /* token_id */, u64 /* epoch */)>,
+    );
+    type Error = ContractError;
+
+    fn __migrate(env: &Env, migration_data: Self::MigrationData) -> Result<(), Self::Error> {
+        let (new_wasm_hash, token_ids_and_epochs) = migration_data;
+
+        for (token_id, epoch) in token_ids_and_epochs {
+            let token_id_config = storage::try_token_id_config(env, token_id.clone())
+                .ok_or(ContractError::InvalidTokenId)?;
+
+            Self::migrate_token_manager(env, token_id_config.token_manager, &new_wasm_hash)?;
+            Self::migrate_interchain_token(env, &token_id, &new_wasm_hash)?;
+            Self::migrate_flow_key(env, &token_id, epoch)?;
```

Comment:
```suggestion
            Self::migrate_flow(env, &token_id, epoch)?;
```
since the enum variant is called `Flow`